### DISCUSSION TITLE
Does srsgnb maintains the mapping between IMSI and RNTI?

### DISCUSSION THREAD

**Discussion by zhuangyufan** [srsran/srsRAN_Project#749] [General Help] [answered]
Dear community,
Does srsgnb maintains the mapping between IMSI and RNTI? If so, where is the mapping maintained?
Thanks a lot.

**Comment by brendan-mcauliffe** [comment, marked as answer]
Hi @egg988, 

There is no relationship between the IMSI and the RNTI, each time a UE attaches the RNTI is changed. For example, if you only connect one UE to the RAN, and disconnect and reconnect again, the RNTI will not be the same.

**Reply by zhuangyufan** [reply]
Thanks for your answer! But my problem is that, after a connection is established, does gNB records the temporary mapping between RNTI and IMSI? Or is this mapping recorded by gNB or core network?

**Reply by pgawlowicz** [reply]
gNB does not even know the IMSI.

**Reply by zhuangyufan** [reply]
Thanks for your answer! May I ask that does 5GC know RNTI? In cellular network, is it necessary to maintain the mapping between RNTI and IMSI? Thanks!

**Reply by herlesupreeth** [reply]
> May I ask that does 5GC know RNTI?

No, 5GC has no idea of the RNTI

> In cellular network, is it necessary to maintain the mapping between RNTI and IMSI?

No, its not necessary

**Reply by zhuangyufan** [reply]
Got it. Thanks a lot!
